### DISCUSSION TITLE
contrl trace by manually

### DISCUSSION THREAD

**Discussion by taotao365s** [apache/skywalking#9488] [Ideas]
I want to control the trace by manually.  I just found there is function  in the SOFA-tracer, so I can contrl the trace.

````
public class DemandThread extends Thread {
        @Override
        public void run() {
            
            //TODO how can we reset new Trace, but not in original trace in before.
            
            while (true) {
                
                //do something.

                try {
                    Thread.sleep(1000 * 20);
                } catch (InterruptedException e) {
                    log.error("error", e);
                }
            }

        }
    }
````

**Comment by wu-sheng** [comment]
If you prefer sofa tracer, you should use them.

**Reply by taotao365s** [reply]
e,,, just want to know how to implement it in skywalking, write one custom plugin? expose CONTEXTMANAGER API to plugin?

**Reply by wu-sheng** [reply]
I don't know what you mean. Plugins always call the core APIs, including ContextManager. 
Have you even read the docs and codes before asking? Asking a project for something from how another project provides is really bad practice. 
**I am super serious about `you should use sofa tracer`, because that is why the world has many open-source projects, rather than one.**
